src/TestFramework/TestFramework/Assertions/Assert.AreEqual.cs

```diff
@@ -2118,3 +2226,121 @@ private static void ThrowAssertAreNotEqualFailed(object? notExpected, object? ac
         ThrowAssertFailed("Assert.AreNotEqual", finalMessage);
     }
 }
+
+internal static class StringPreviewHelper
+{
+    public static Tuple<string, string, int> CreateStringPreviews(string expected, string actual, int diffIndex, int fullPreviewLength)
+    {
+        int ellipsisLength = 3; // Length of the ellipsis "..."
+
+        if (fullPreviewLength % 2 == 0)
+        {
+            throw new ArgumentException($"{nameof(fullPreviewLength)} must be odd, but it was even.");
+        }
+
+        if (fullPreviewLength < 9)
```

Comment:
should these be named constants?

Reply:
why 9? this feels surprising, including the length of 41 by the caller

Reply:
added better comment, keeping without some constant since this is used in 1 place only and the name does not add any more info.